### PR TITLE
#5153 Add route test for 404 and 501 pages

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist

- [x] [Review Contribution Guidelines](https://oharastream.github.io/en/docs/master/contributing/)
- [x] Add at least one reviewer (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add an assignee (of course, you are the assignee by default)
- [x] Do not set milestone, projects and labels (PR labels are added automatically by Jenkins)
- [x] Link to an issue
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (If the failed tests are unrelated to this PR, add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/cypress/it/appBar/appBar.test.ts [hunk 1]
```diff
@@ -170,4 +170,66 @@ describe('App Bar', () => {
...
+        onBeforeLoad(win) {
+          cy.stub(win, 'open').as('openStub');
+        },
+      })
+        .contains('501')
+        .should('exist');
+
+      cy.findByText('SEE AVAILABLE RELEASES').click();
+      cy.get('@openStub')
+        .invoke('calledWith', 'https://github.com/oharastream/ohara/releases')
```
Comment (saivirtue) [inline ohara-manager/client/cypress/it/appBar/appBar.test.ts:217]: A way to assert we opened a new window with specific url

File: ohara-manager/client/cypress/it/appBar/appBar.test.ts [hunk 2]
```diff
@@ -170,4 +170,66 @@ describe('App Bar', () => {
...
+  context('Not found page', () => {
+    it('should display not found page', () => {
+      cy.visit('/jladkf/safkj/ksjdl/jlkfsd/kjlfds')
+        .contains('404')
+        .should('exist');
+
+      cy.findByText('BACK TO HOME').click();
+      cy.location('pathname').should(
+        'not.equal',
+        '/jladkf/safkj/ksjdl/jlkfsd/kjlfds',
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/appBar/appBar.test.ts:231]: What about '/' :)

File: ohara-manager/client/cypress/it/appBar/appBar.test.ts [hunk 3]
```diff
@@ -170,4 +170,66 @@ describe('App Bar', () => {
       cy.location('pathname').should('equal', `/${name}`);
     });
   });
+
+  context('Redirect routes', () => {
+    it('should redirect to default workspace and pipeline', () => {
+      // We have two workspaces now
+      // it should redirect one of them by default
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/appBar/appBar.test.ts:177]: _to one of..._

File: ohara-manager/client/cypress/it/appBar/appBar.test.ts [hunk 4]
```diff
@@ -170,4 +170,63 @@ describe('App Bar', () => {
...
+  });
+
+  context('Not found page', () => {
+    it('should display not found page', () => {
+      cy.visit('/jladkf/safkj/ksjdl/jlkfsd/kjlfds')
+        .contains('404')
+        .should('exist');
+
+      cy.findByText('BACK TO HOME').click();
+      cy.location('pathname').should('not.equal', '/');
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/appBar/appBar.test.ts:229]: Sam, it should be -> should("equal", '/') instead of should('not.equal', '/')

Comment (saivirtue) [reply ohara-manager/client/cypress/it/appBar/appBar.test.ts:229]: We have workspaces in this test, so the url should "not" be "/" :)

Comment (saivirtue) [reply ohara-manager/client/cypress/it/appBar/appBar.test.ts:229]: But the purpose of this assertion is not clear... I will do some refactor about this.

Comment (konekoya) [reply ohara-manager/client/cypress/it/appBar/appBar.test.ts:229]: >We have two workspaces in this test

Ah, that's why 😅

**Pull request merged**